Libraries/LibWeb/HTML/HTMLButtonElement.cpp

```diff
@@ -153,9 +155,108 @@ void HTMLButtonElement::activation_behavior(DOM::Event const& event)
             return;
     }
 
-    // FIXME: 4. Let target be the result of running element's get the commandfor associated element.
-    // FIXME: 5. If target is not null:
-    //           ...
+    // 4. Let target be the result of running element's get the commandfor associated element.
+    //    AD-HOC: Target needs to be an HTML Element in the following steps.
+    GC::Ptr<HTMLElement> target = as<HTMLElement>(m_command_for_element.ptr());
```

Comment:
If it's possible for `m_command_for_element` to be a `DOM::Element` that isn't a HTML element, then it would be preferable to adapt the steps below to work with all kinds of `DOM::Element`.

Reply:
The popover related function in steps 9 to 11 only work with `HTMLElement` and this is also described in the spec. I replaced `as` with `as_if` in order to prevent crashes. I didn't know `as_if` before you mentioned it😅